src/pages/LogInWithShortLivedTokenPage.js

```diff
@@ -73,18 +74,18 @@ class LogInWithShortLivedTokenPage extends Component {
 
         // exitTo is URI encoded because it could contain a variable number of slashes (i.e. "workspace/new" vs "workspace/<ID>/card")
         const exitTo = decodeURIComponent(lodashGet(this.props.route.params, 'exitTo', ''));
+
+        // We have two routes in our history, "/" and "transition". We want to dismiss the trasition route, remove it from the history
+        // or else a user could be routed back to `/transition` when they close the modal.
```

Comment:
>or else a user could be routed back to `/transition` when they close the modal.

What are the steps to reproduce this and which modal we are referring to in this comment?

Reply:
This is all that was said in the previous comment:
```
// In order to navigate to a modal, we first have to dismiss the current modal. But there is no current
        // modal you say? I know, it confuses me too. Without dismissing the current modal, if the user cancels out
        // of the workspace modal, then they will be routed back to
        // /transition/<accountID>/<email>/<authToken>/workspace/<policyID>/card and we don't want that. We want them to go back to `/`
        // and by calling dismissModal(), the /transition/... route is removed from history so the user will get taken to `/`
        // if they cancel out of the new workspace modal.
```

You can reproduce this error by commenting out the `dismissModal()` on [this line](https://github.com/Expensify/App/blob/d5ea6921054e31b3f8919fc61774c1bab318b19b/src/pages/LogInWithShortLivedTokenPage.js#L82). 
1. Have an account and choose track plan (get redirected and go back to old dot)
2. You now will see this inbox task which will take you to the `bank-account` modal in new dot. Click "continue setup"
    ![image](https://user-images.githubusercontent.com/12980144/146314933-2e926212-c247-46ca-8903-73dfb5eabf8d.png)


E.g. of what the comments say will happen
![Pg7Peh2Wty](https://user-images.githubusercontent.com/12980144/146314916-899f0b14-e271-4869-807a-32402f57881e.gif)

Reply:
What if we changed this to something like 

>When the user lands on this page they should have two routes in their history - the main route and `LoginWithShortLivedTokenPage`. We need to dismiss this page after we are done creating a policy and when navigating the user elsewhere or else they will see a persistent loading spinner.

Reply:
>or else a user could be routed back to `/transition` when they close the modal.

Basically, I don't think this is true. They don't get "routed back" they just never left in the first place.

Reply:
Yep that makes sense